### PR TITLE
Add a notification about BuildCheck enabled for the current build

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
Fixes #[10135](https://github.com/dotnet/msbuild/issues/10135)

### Context
The users have to be notified if BuildCheck got invoked during the build.

### Changes Made
The message got logged once the logger service is available and custom analyzers are defiled:
image

### CODE REVIEW

**Comment by JanKrivanek** [comment]
The original requirement from the security review was to be able to locate all the check names (and respective package names that contained them) that were enabled in the build, despite the checks might not have produced any outputs.

Sample simplisitc possible output:

```
The BuildCheck is enabled for this build. Following Checks were enabled on at least one of the projects:
| name | source |
| SharedOutputPathsCheck | builtin |
| ContosoProjectStructureCheck | Contoso.BuildChecks 0.0.2 |
```

File: NuGet.config [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <add key="dotnet8" value="https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet8/nuget/v3/index.json" />
     <add key="dotnet8-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8-transport/nuget/v3/index.json" />
     <add key="BuildXL" value="https://pkgs.dev.azure.com/ms/BuildXL/_packaging/BuildXL/nuget/v3/index.json" />
+    <add key="local" value="C:\mbuild\ykovalova_msbuild\msbuild\artifacts\bin\Microsoft.Build.BuildCheck.UnitTests\CustomAnalyzers" />
```
Comment (JanKrivanek) [inline NuGet.config:12]: Intentional?

File: src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs [hunk 2]
```diff
@@ -52,7 +52,8 @@ public List<BuildAnalyzerFactory> CreateBuildAnalyzerFactories(AnalyzerAcquisiti
 
             if (availableTypes.Count != analyzerTypes.Count)
             {
-                availableTypes.Except(analyzerTypes).ToList().ForEach(t => _loggingService.LogComment(buildEventContext, MessageImportance.Normal, "CustomAnalyzerBaseTypeNotAssignable", t.Name, t.Assembly));
+                availableTypes.Except(analyzerTypes).ToList()
+                    .ForEach(t => _loggingService.LogComment(buildEventContext, MessageImportance.Normal, "CustomAnalyzerBaseTypeNotAssignable", t.Name, t.Assembly));
```
Comment (JanKrivanek) [inline src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs:56]: Are we able to get the package name as well?

Comment (YuliiaKovalova) [reply src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs:56]: we operate with dlls directly. I can have a location reported.

Comment (YuliiaKovalova) [reply src/Build/BuildCheck/Acquisition/BuildCheckAcquisitionModule.cs:56]: it's already there
image

**Comment by YuliiaKovalova** [comment]
> The original requirement from the security review was to be able to locate all the check names (and respective package names that contained them) that were enabled in the build, despite the checks might not have produced any outputs.
> 
> Sample simplisitc possible output:
> 
> ```
> The BuildCheck is enabled for this build. Following Checks were enabled on at least one of the projects:
> | name | source |
> | SharedOutputPathsCheck | builtin |
> | ContosoProjectStructureCheck | Contoso.BuildChecks 0.0.2 |
> ```

done, in a different way.

**Review by JanKrivanek — APPROVED** [review]
:shipit: ;-)

File: src/Build/Resources/Strings.resx [hunk 3]
```diff
@@ -2130,6 +2130,12 @@ Utilization:          {0} Average Utilization: {1:###.0}</value>
   <data name="ProjectImportsCollectorFileIOFail" xml:space="preserve">
     <value>An exception occurred when adding the file '{0}', Exception: '{1}'</value>
   </data>
+  <data name="BuildCheckEnabled" xml:space="preserve">
+    <value>The BuildCheck is enabled for this build.</value>
+  </data>
+  <data name="CustomAnalyzerRegistered" xml:space="preserve">
```
Comment (surayya-MS) [inline src/Build/Resources/Strings.resx:2136]: I don't see where this is used. Probably was left after changes?

**Review by surayya-MS — APPROVED** [review]
Looks good to me! 
I will resolve conflicts that popped up because of BuildCheck Replay Mode

**Review by maridematte — APPROVED** [review]

File: src/Build/BackEnd/BuildManager/BuildManager.cs [hunk 4]
```diff
@@ -577,6 +577,12 @@ public void BeginBuild(BuildParameters parameters)
                 // Log deferred messages and response files
                 LogDeferredMessages(loggingService, _deferredBuildMessages);
 
+                // Log if BuildCheck is enabled
+                if (_buildParameters.IsBuildCheckEnabled)
+                {
+                    loggingService.LogComment(buildEventContext: BuildEventContext.Invalid, MessageImportance.Normal, "BuildCheckEnabled");
```
Comment (surayya-MS) [inline src/Build/BackEnd/BuildManager/BuildManager.cs:583]: @JanKrivanek, @YuliiaKovalova we should probably do same for replaying binlog, right? We can do this in a different PR.

Comment (JanKrivanek) [reply src/Build/BackEnd/BuildManager/BuildManager.cs:583]: Good point - it's good to have a feature parity.

It's not a high-pri though - as this was requested (for the security reasons) to inform about possible pluggability logic engagging as a part of the build process.
